api/node/index.ts

```diff
@@ -4,40 +4,88 @@
 import * as path from "path";
 
 import * as napi from "./rust-module";
-export { Diagnostic, DiagnosticLevel, Window, Brush, Color, ImageData, Point, Size, SlintModelNotify } from "./rust-module";
+export { Diagnostic, DiagnosticLevel, Brush, Color, SlintModelNotify } from "./rust-module";
 
 /**
- * ModelPeer is the interface that the run-time implements. An instance is
- * set on dynamic {@link Model} instances and can be used to notify the run-time
- * of changes in the structure or data of the model.
+ *  Represents a two-dimensional point.
  */
-export interface ModelPeer {
+export interface Point {
+    x: number,
+    y: number
+}
+
+/**
+ *  Represents a two-dimensional size.
+ */
+export interface Size {
+    width: number,
+    height: number
+}
+
+/**
+ * This type represents a window towards the windowing system, that's used to render the
+ * scene of a component. It provides API to control windowing system specific aspects such
+ * as the position on the screen.
+ */
+export interface Window {
     /**
-     * Call this function from our own model to notify that fields of data
-     * in the specified row have changed.
-     * @argument row
+     * Shows the window on the screen. An additional strong reference on the
+     * associated component is maintained while the window is visible.
      */
-    rowDataChanged(row: number): void;
+    show(): void;
+
+    /** Hides the window, so that it is not visible anymore. */
+    hide(): void;
+
     /**
-     * Call this function from your own model to notify that one or multiple
-     * rows were added to the model, starting at the specified row.
-     * @param row
-     * @param count
+     * Returns the visibility state of the window. This function can return false even if you previously called show()
+     * on it, for example if the user minimized the window.
+     */
+    get isVisible(): boolean;
+
+    /** Returns the logical position of the window on the screen. */
+    get logicalPosition(): Point;
+
+    /** Sets the logical position of the window on the screen. */
+    set logicalPosition(position: Point);
+
+    /** Returns the physical position of the window on the screen. */
+    get physicalPosition(): Point;
+
+    /** Sets the physical position of the window on the screen. */
+    set physicalPosition(position: Point);
+
+    /** Returns the logical size of the window on the screen, */
+    get logicalSize(): Size;
+
+    /** Sets the logical size of the window on the screen, */
+    set logicalSize(size: Size);
+
+    /** Returns the physical size of the window on the screen, */
+    get physicalSize(): Size;
+
+    /** Sets the logical size of the window on the screen, */
+    set physicalSize(size: Size);
+}
+
+/**
+ * An image data type that can be displayed by the Image element.
```

Comment:
In the review we said "link to https://developer.mozilla.org/en-US/docs/Web/API/ImageData in the docs" :)

Reply:
done.